backend/lib/endpoints/posts.js

```diff
@@ -207,19 +216,49 @@ async function routes(app) {
           },
         },
       ];
+      /* eslint-enable sort-keys */
+
+      const aggregationPipelineResults = [
+        ...sortAndFilterSteps,
+        ...paginationSteps,
+        ...lookupSteps,
+        ...projectionSteps,
+      ];
+
+      // Get the total results without pagination steps but with filtering aplyed - totalResults
+      /* eslint-disable sort-keys */
+      const totalResultsAggregationPipeline = await Post.aggregate([
+        ...sortAndFilterSteps,
+        ...lookupSteps,
```

Comment:
```suggestion
```

We don't need to include lookupSteps for total - we really just need the filters. The sort is also unnecessary - and at scale but create a drag on performance. Not sure if MongoDB is smart enough to ignore them since they are not necessary for the result but I doubt it. 

We really just need `$match: { $and: filters } }` plus the grouping.